espnet/tts/pytorch_backend/tts.py

```diff
@@ -202,7 +200,7 @@ def train(args):
     # reverse input and output dimension
     idim = int(valid_json[utts[0]]['output'][0]['shape'][1])
     odim = int(valid_json[utts[0]]['input'][0]['shape'][1])
-    if args.use_cbhg:
+    if args.use_cbhg if hasattr(args, "use_cbhg") else False:
```

Comment:
I do not like this. We can capsule these `use_cbhg` blocks in `TTSInterface`. How about this?
```python
class TTSInterface:
    ...
    def setup_dim(self, spc_dim, spk_embed_dim):
        """setup data dependent dim

        :param int spc_dim: target spectrum dim
        :param int spk_embed_dim: input speaker embedding dim
        """
        pass

    @property
    def use_second_target(self):
        """this property is called to instantiate LoadInputsAndTargets

        :rtype bool:
        """
        return False

    def register_plot_keys(self, plot_keys):
        """register keys to plot during training. keys should match what `chainer.reporter` reports

        :param list[str] plot_keys: keys to plot during training
        """
        pass
```

```python
class Tacotron2(TTSInterface, torch.nn.Module):
    ...
    def setup_dim(self, spc_dim, spk_embed_dim):
        self.spk_embed_dim = spk_embed_dim
        if self.use_cbhg:
            self.spc_dim = args.spc_dim

    @property
    def use_second_target(self):
        return False

    def register_plot_keys(self, plot_keys):
        plot_keys += ['main/cbhg_l1_loss', 'validation/main/cbhg_l1_loss',
                              'main/cbhg_mse_loss', 'validation/main/cbhg_mse_loss']
```

Reply:
For `set_dim(...)`, it is better to set these args in `run.sh` or `conf/train_tacotron.yaml` if we can.

Reply:
I decide to use `use_second_target` as argment because if we set it as model property, it becomes dirty due to the `DataParallel`.

Reply:
cool